### PR TITLE
Use RNG HW rendering

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Use RNG HW rendering.

It probably also increases the justification of https://github.com/google/horologist/issues/1221, as we don't have a known date for AGP screenshotTest to save us.

#### WHY

Resolve https://github.com/robolectric/robolectric/issues/8592

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: .github/workflows/fixup.yml
Comment (yschimke) [inline .github/workflows/fixup.yml:1]: To revert

File: roboscreenshots/src/main/java/com/google/android/horologist/screenshots/ScreenshotTestRule.kt [hunk 1]
```diff
@@ -283,7 +295,7 @@ public class ScreenshotTestRule(
             public var enableA11y: Boolean = false
             public var screenTimeText: @Composable () -> Unit = defaultScreenTimeText()
             public var testLabel: String? = null
-            public var record: RecordMode = RecordMode.fromProperty(System.getProperty("screenshot.record"))
+            public var record: RecordMode = RecordMode.Record
```
Comment (yschimke) [inline roboscreenshots/src/main/java/com/google/android/horologist/screenshots/ScreenshotTestRule.kt:298]: TODO revert

File: composables/src/test/snapshots/images/com.google.android.horologist.composables_DatePickerA11yTest_interactionTest_2.png
Comment (yschimke) [inline composables/src/test/snapshots/images/com.google.android.horologist.composables_DatePickerA11yTest_interactionTest_2.png:1]: Hmmm.....

File: sample/src/test/snapshots/images/com.google.android.horologist.screensizes_SectionedListTest_screenshot[7]_largedevicesmallfonts.png
Comment (yschimke) [inline sample/src/test/snapshots/images/com.google.android.horologist.screensizes_SectionedListTest_screenshot[7]_largedevicesmallfonts.png:1]: This is the core of the change.

File: roboscreenshots/src/main/java/com/google/android/horologist/screenshots/ScreenshotTestRule.kt [hunk 2]
```diff
@@ -174,6 +177,54 @@ public class ScreenshotTestRule(
...
+            ClipMode.None -> false
+        }
+        return if (shouldClip) {
+            circularClip(bitmap)
+        } else {
+            bitmap
+        }
+    }
+
+    fun circularClip(image: Bitmap): Bitmap {
```
Comment (yschimke) [inline roboscreenshots/src/main/java/com/google/android/horologist/screenshots/ScreenshotTestRule.kt:212]: With hardware renderer, there is no transparent regions, so need to clip afterwards to match.

File: auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png
Comment (yschimke) [inline auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png:1]: Interesting size change.  Maybe just wasn't failing before?

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png
Comment (yschimke) [inline compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png:1]: TODO explain this...

Comment (yschimke) [reply compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png:1]: I think these are correct.  When mirrored + rtl, it goes back to the default.

Comment (yschimke) [reply compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png:1]: Maybe target sdk 33 change?

File: roboscreenshots/src/main/java/com/google/android/horologist/screenshots/RobolectricTempHelpers.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    @RequiresApi(Build.VERSION_CODES.Q)
+    private fun generateBitmapUsingHardwareRenderNative(view: View, destBitmap: Bitmap) {
+        val width = view.width
+        val height = view.height
+        val imageReader = ImageReader.newInstance(width, height, PixelFormat.RGBA_8888, 1)
+        val renderer = HardwareRenderer()
+        renderer.setSurface(imageReader.surface)
+        val nativeImage: Image = imageReader.acquireNextImage()
+
+//        setupRendererShadowProperties(renderer, view)
```
Comment (luizgrp) [inline roboscreenshots/src/main/java/com/google/android/horologist/screenshots/RobolectricTempHelpers.kt:106]: any particular reason to comment it out?

File: auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png
Comment (luizgrp) [inline auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png:1]: The icon size has increased. Based on the code changes in this PR, I'm assuming it's only a side effect in the screenshots and not when running the app?

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png
Comment (luizgrp) [inline compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png:1]: Here the icon is not mirrored when it should. But it makes me think if this actually exposing an issue either:
- with our Icon implementation, when receiving "AutoMirrored" icons, it should not scale.
- with our test code, when using "AutoMirrored" icons, it should not pass `iconRtlMode = IconRtlMode.Mirrored`.

Either way, we should check the results when running an app with a device in rtl.

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_IconTest_autoMirroredRtl.png
Comment (luizgrp) [inline compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_IconTest_autoMirroredRtl.png:1]: At first I would expect to see the same issue as in the screenshot from my previous comment. However, I see that this test is not passing `rtlMode = IconRtlMode.Mirrored`. I think two more tests should be added to IconTest, duplicating `autoMirroredLTR` and this one, but adding `rtlMode = IconRtlMode.Mirrored`.

**Review by luizgrp — APPROVED** [review]
LGTM - but we should understand what's going on with the screenshots in RTL

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png
Comment (luizgrp) [reply compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonTest_mirroredRtl.png:1]: sorry, missed your comments before reviewing

File: roboscreenshots/src/main/java/com/google/android/horologist/screenshots/RobolectricTempHelpers.kt (hunk 3 above)
Comment (yschimke) [reply roboscreenshots/src/main/java/com/google/android/horologist/screenshots/RobolectricTempHelpers.kt:106]: It comes from the original source.  But has android.os.internal.R constants that I don't have. Doesn't seem an issue, maybe relates to multi windows?

File: auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png
Comment (yschimke) [reply auth/composables/src/test/snapshots/images/com.google.android.horologist.auth.composables.dialogs_SignedInConfirmationDialogTest_signedInConfirmationDialog.png:1]: I'll check.

**Comment by yschimke** [comment]
Too big to iterate on, going to land since there are no main code changes.  Only tests.  Will work on two issues and feedback from a previous PR.

**Pull request merged**